types/index.d.ts

```diff
@@ -0,0 +1,160 @@
+import type { AedesPacket } from 'aedes-packet';
+import type { IncomingMessage } from 'http';
+import type { Socket } from 'net';
+import type { Duplex, Readable } from 'stream';
+import type { EventEmitter } from 'events';
+
+export type { AedesPacket as Packet } from 'aedes-packet';
+
+export interface AedesPersistenceSubscription {
+  clientId?: string;
+  topic: string;
+  qos: number;
+}
+
+export interface Brokers {
+  [brokerId: string]: {
+    brokerId: string;
+  };
+}
+
+type Connection = Duplex | Socket;
+
+interface Client extends EventEmitter {
+  id: string;
+  clean: boolean;
+  version: number;
+  conn: Connection;
+  req?: IncomingMessage;
+  connecting: boolean;
+  connected: boolean;
+  closed: boolean;
+}
+
+declare class MemoryPersistence {
+  private _retained: any[];
+  private _subscriptions: Map<string, AedesPersistenceSubscription>;
+  private _clientsCount: number;
+  private _trie : any
+  private _outgoing: Record<string, AedesPacket>;
+  private _incoming: Record<string, AedesPacket>;
+  private _wills: Record<string, any>;
```

Comment:
Also wills are like aedes packets, if there are additiona lprops you could define new types like `WillPacket` and use the `&` operator to add the missing props